src/analysis/functions.rs

```diff
@@ -137,6 +137,89 @@ pub fn analyze<F: Borrow<library::Function>>(
     funcs
 }
 
+fn fixup_special_functions(
+    env: &Env,
+    name: &str,
+    type_tid: library::TypeId,
+    parameters: &mut Parameters
+) {
+    // Workaround for some _hash() / _compare() / _equal() functions taking
+    // "gconstpointer" as arguments instead of the actual type
+    if name == "hash" && parameters.c_parameters.len() == 1 {
+        if parameters.c_parameters[0].c_type == "gconstpointer" {
+            use analysis::ffi_type;
+
+            parameters.rust_parameters[0].typ = type_tid;
+            parameters.c_parameters[0].typ = type_tid;
+            parameters.c_parameters[0].instance_parameter = true;
+            parameters.c_parameters[0].ref_mode = RefMode::ByRef;
+            parameters.c_parameters[0].transfer = Transfer::None;
+            let glib_name = env.library.type_(type_tid).get_glib_name().unwrap();
+            let ffi_name = ffi_type::fix_name(env, type_tid, &glib_name).unwrap();
+            parameters.transformations[0] = Transformation {
+                ind_c: 0,
+                ind_rust: Some(0),
+                transformation_type: TransformationType::ToGlibPointer {
+                    name: parameters.rust_parameters[0].name.clone(),
+                    instance_parameter: true,
+                    transfer: Transfer::None,
+                    ref_mode: RefMode::ByRef,
+                    to_glib_extra: String::new(),
+                    explicit_target_type: format!("*mut {}", ffi_name),
```

Comment:
Maybe better use "*const" here, as we IMHO always implemented it too

Reply:
and maybe use this only if function in trait

Reply:
> Maybe better use "*const" here, as we IMHO always implemented it too

I think that's not implemented for `Shared`, not sure. Will check

Reply:
Not implemented for `Object` at least, so let's keep it `*mut`